.storybook/webpack.config.js

```diff
@@ -148,11 +149,12 @@ module.exports = async ({ config, mode }) => {
       test: /\.(ts|tsx)$/,
       loader: 'ts-loader',
       options: {
+        transpileOnly: !isCI,
```

Comment:
How about we just specify this via a cli / environment variable? It would solve two issues:
1. Different ways to detect different CI environments other than the jenkins-specific vars in `ci.ts`
2. Being able to run with checking on local builds when trying to fix errors

Reply:
I agree with @petemill. This should be explicit. My preferred approach (if feasible) would be via a separate command that does nothing else but run the type checks, similarly to `npm run presubmit`. Otherwise, via a CLI argument. If that is not possible either, then via an environment variable whose sole purpose is to turn the feature on or off. For example, `TRANSPILE_ONLY`. I can make the necessary changes in CI.

I also want to explain why I have the above preference. In short, it is because explicit is better than implicit. There are several ways in which the current implementation can cause trouble, just because it is implicit:

1. We are planning to switch from Jenkins to TeamCity. Once that happens, the code proposed here will no longer work. But we won't even notice! Failing type checks will simply not show up as errors in CI. This is because the current code encodes knowledge about Jenkins. It's really not something it should do.
2. Like @petemill said, what if I do want to run the check synchronously locally? Then I have to fake Jenkins' environment variables on my local machine. That's not just awkward but also counter-intuitive.

I was personally affected by a problem caused by an implicit implementation in related code in https://github.com/brave/brave-browser/issues/34984. It was confusing and cost me time to debug. I believe that every implicit solution can be turned into an explicit one, maybe with a very small amount of additional manual one-time effort on the part of the developer. But it avoids painful and unnecessary debugging sessions.

I like the Zen of Python, which also talks about explicitness:

```
Beautiful is better than ugly.
Explicit is better than implicit.
Simple is better than complex.
Complex is better than complicated.
Flat is better than nested.
Sparse is better than dense.
Readability counts.
Special cases aren't special enough to break the rules.
Although practicality beats purity.
Errors should never pass silently.
Unless explicitly silenced.
In the face of ambiguity, refuse the temptation to guess.
There should be one-- and preferably only one --obvious way to do it.[c]
Although that way may not be obvious at first unless you're Dutch.
Now is better than never.
Although never is often better than right now.[d]
If the implementation is hard to explain, it's a bad idea.
If the implementation is easy to explain, it may be a good idea.
Namespaces are one honking great idea – let's do more of those!
```